pkg/flow/pubsub/pubsub.go

```diff
@@ -369,7 +376,9 @@ func (pubsub *Pubsub) flush() {
 
 func (pubsub *Pubsub) timeFlush() {
 	pubsub.bufferMtx.Lock()
-	go pubsub.flush()
+	// defer pubsub.bufferMtx.Unlock()
+
+	pubsub.flush()
```

Comment:
@alankm I removed the goroutine here since it occasionally caused panics under server load. However, the periodicFlush method is still executed via a goroutine. I think it was safe to do so.

Also, I think we should continue working towards finally getting rid of this old pub-sub system.

Reply:
What the... 

I just had a look at the original for comparison and this goroutine makes no sense. I think the mutex lock above it literally guarantees that it runs synchronously anyway...

Reply:
the panic i encountered was due to locking a locked mutex. 
but i agree there was no need in having  this to be a go routine